File: Programar em Java/11-Listas.java
// YouTube: https://youtu.be/VFwjISUooB4
// Publicação: https://caffeinealgorithm.com/blog/20220218/listas-em-java/

package com.caffeinealgorithm.programaremjava;

import java.util.ArrayList;
import java.util.List;

public class Listas {
  public void Run() {
    List<String> cores = new ArrayList<>();

    cores.add("Azul");
    cores.add("Verde");
    cores.add("Amarelo");
    cores.add("Vermelho");
    cores.add("Laranja");

    cores.remove("Laranja");
    // cores.clear();

    System.out.printf("Número de cores: %d\n", cores.size()); // Número de cores: 4
    System.out.printf("Primeira cor: %s\n", cores.get(0)); // Primeira cor: Azul
    System.out.printf("Última cor: %s", cores.get(cores.size() - 1)); // Última cor: Vermelho
  }
}

File: Programar em C#/11-Listas.cs
// YouTube: https://youtu.be/Eov-gcwg9zc
// Publicação: https://caffeinealgorithm.com/blog/listas-em-csharp/

using System;
using System.Collections.Generic;
using System.Linq;

namespace Base {
  class Listas {
    public void Run() {
      List<string> cores = new List<string>();

      cores.Add("Azul");
      cores.Add("Verde");
      cores.Add("Amarelo");
      cores.Add("Vermelho");
      cores.Add("Laranja");

      cores.Remove("Laranja");
      cores.Reverse();
      cores.Sort();
      // cores.Clear();

      Console.WriteLine($"Número de cores: {cores.Count}"); // Número de cores: 4
      Console.WriteLine($"Primeira cor: {cores.First()}"); // Primeira cor: Amarelo
      Console.WriteLine($"Última cor: {cores.Last()}"); // Última cor: Vermelho
    }
  }
}
